enterprise/internal/a8n/service.go

```diff
@@ -297,14 +297,22 @@ func RunChangesetJob(
 	// We should probably persist the `headRefName` on `ChangesetJob` and keep
 	// it stable across retries and only set it the first time.
 
+	ensureUniqueRef := true
+	if job.Branch == "" {
```

Comment:
Doesn't this have to be this?

```suggestion
	if job.Branch != "" {
```